api/src/services/dwc-service.ts

```diff
@@ -26,17 +59,23 @@ export class DwCService extends DBService {
    * @memberof DwCService
    */
   async decorateTaxonIDs(jsonObject: Record<any, any>): Promise<Record<any, any>> {
-    const taxonomyService = new TaxonomyService();
-
     // Find and return all nodes that contain `taxonID`
     const pathsToPatch = JSONPath({ path: '$..[taxonID]^', json: jsonObject, resultType: 'all' });
 
+    // Collect all unique taxon IDs
+    const taxonIds = this.collectTaxonIDs(pathsToPatch);
+
+    // Make a request for each unique taxon ID
+    // TODO this approach assumes every row in the provided has a unique taxon ID
```

Comment:
super-nit: I think this is worded wrong? I think you mean, assumes every row is not unique (that the unique set of taxoniIDs is small).

Reply:
updated the text